### DISCUSSION TITLE
Renaming branches

### DISCUSSION THREAD

**Discussion by eliottwiener** [newren/git-filter-repo#301] [Q&A] [answered]
[In the README there is an example given](https://github.com/newren/git-filter-repo#solving-this-with-filter-repo) that uses `--to-subdirectory-filter` with `--tag-rename`:

```sh
git filter-repo --path src/ --to-subdirectory-filter my-module --tag-rename '':'my-module-'
```

Is there a flag for renaming branches as well, similar to `--tag-rename`?

I would like to be able to rename all branches to add a prefix.

**Comment by newren** [comment, marked as answer]
If you want to rename a branch, just use `git branch -m <oldbranch> <newbranch>`; no need to use a history filtering tool to achieve that.  You'll note that `git tag` does not have a rename operation similar to `git branch -m ...`.  The reason is that there are special tag objects (but not any special branch objects), and those tag objects not only store a message but also (semi-redundantly) record the name of the tag.  For the non-lightweight tags to really be renamed, those tag objects also have to be rewritten to record the new name, which is what makes it a history modifying operation.

**Reply by eliottwiener** [reply]
Thank you for the answer. I appreciate the additional explanation on why `--tag-rename` exists within the scope of git-filter-repo.

**Reply by gasbi** [reply]
Just for completion, you can use grep to rename all branches at the same time because with `git branch -m` seems that you have to do it one by one.
`git branch | grep "StringToFind" | awk '{original=$1; sub("StringToFind","ReplaceWith"); print original, $1}' | xargs -n 2 git branch -m`

Taken from https://stackoverflow.com/a/45118832/4566352
